app/models/user_seen_subject.rb

```diff
@@ -14,4 +14,8 @@ def add_subjects(subjects)
     subject_ids.concat(subjects)
     save!
   end
+
+  def already_seen?(ids)
```

Comment:
can you change this method name to be not_seen or something? Looks like thats what it's actually doing.

Also seems the set intersection for large seen sets will be slower than just iterating over the classification subject ids and searching the seen set via `include?` https://gist.github.com/camallen/29a63c16b03bb9caa255